code/modules/food_and_drinks/kitchen_machinery/monkeyrecycler.dm

```diff
@@ -47,21 +53,25 @@
 	default_deconstruction_crowbar(O)
 
 	if(istype(O, /obj/item/multitool))
-		cycle_through++
-		switch(cycle_through)
-			if(1)
-				cube_type = /obj/item/reagent_containers/food/snacks/monkeycube/farwacube
-			if(2)
-				cube_type = /obj/item/reagent_containers/food/snacks/monkeycube/wolpincube
-			if(3)
-				cube_type = /obj/item/reagent_containers/food/snacks/monkeycube/stokcube
-			if(4)
-				cube_type = /obj/item/reagent_containers/food/snacks/monkeycube/neaeracube
-			if(5)
-				cube_type = /obj/item/reagent_containers/food/snacks/monkeycube
-				cycle_through = 0
-		to_chat(user, "<span class='notice'>You change the monkeycube type to [initial(cube_type.name)].</span>")
-
+		if(!panel_open)
+			cycle_through++
+			switch(cycle_through)
+				if(1)
+					cube_type = /obj/item/reagent_containers/food/snacks/monkeycube/farwacube
+				if(2)
+					cube_type = /obj/item/reagent_containers/food/snacks/monkeycube/wolpincube
+				if(3)
+					cube_type = /obj/item/reagent_containers/food/snacks/monkeycube/stokcube
+				if(4)
+					cube_type = /obj/item/reagent_containers/food/snacks/monkeycube/neaeracube
+				if(5)
+					cube_type = /obj/item/reagent_containers/food/snacks/monkeycube
+					cycle_through = 0
+			to_chat(user, "<span class='notice'>You change the monkeycube type to [initial(cube_type.name)].</span>")
+		else
+			var/obj/item/multitool/M = O
+			M.buffer = src
+			to_chat(user, "<span class='notice'>You save the data in the [O.name]'s buffer.</span>")
```

Comment:
`to_chat(user, "<span class='notice'>You log [src] in the multitool's buffer.</span>")`